src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,46 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.DataProcessingException;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class MovieServiceImpl implements MovieService {
+    private final SessionFactory factory;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
+        this.factory = sessionFactory;
+    }
+
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error adding Movie: " + movie, e);
+        } finally {
+            assert session != null;
```

Comment:
Using `assert` to check if the session is not null is not recommended for production code. Consider using a more robust null check or handling mechanism to ensure the session is properly closed.